test/sys/test_pthread.rs

```diff
@@ -20,3 +20,16 @@ fn test_pthread_kill_none() {
     pthread_kill(pthread_self(), None)
         .expect("Should be able to send signal to my thread.");
 }
+
+#[test]
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+fn test_pthread_name_np() {
+    let tid = pthread_self();
+    let name = b"nix-rust";
```

Comment:
The fact that it takes three lines to put `name` into the right format suggests that the function's interface isn't well-designed.  The goal is to be able to call it in any convenient way, like this:
```rust
pthread_setname_np(tid, "foo");    // Rust string
pthread_setname_np(tid, b"foo");    // Byte vector
pthread_setname_np(tid, CString::new("foo"));    // C String
```